src/codegen/bitfield_unit.rs

```diff
@@ -100,3 +100,21 @@ where
         }
     }
 }
+
+#[inline]
+pub fn signed_val(val: u64, bits: usize) -> i64 {
+    if bits == 64 {
```

Comment:
Instead of having this here, let's have it as a static function in the type, to minimize the chances of naming conflicts (so put it inside the `impl` block, and call it like `Self::signed_val(..)` instead.